### ISSUE TITLE
NoClassDefFoundError when selecting spring-data-jdbc feature

### ISSUE THREAD

**Issue by johncurrier** [micronaut-projects/micronaut-starter#1467]
### Description

The Micronaut app generated with the spring-data-jdbc feature fails to run due to
  `java.lang.NoClassDefFoundError: io/micronaut/transaction/hibernate5/MicronautSessionContext`

### Steps to Reproduce

1. Configure a Java 11 starter app with gradle and spring-data-jdbc: https://micronaut.io/launch?type=DEFAULT&name=demo-jdbc&package=com.example&javaVersion=JDK_11&lang=JAVA&build=GRADLE&test=JUNIT&features=spring-data-jdbc&version=3.7.1
2. Unzip it.
3. Attempt to run it via ./gradlew run

### Expected Behaviour

It should start w/o exceptions.

### Actual Behaviour

Receive an exception ending with:
`Caused by: java.lang.ClassNotFoundException: io.micronaut.transaction.hibernate5.MicronautSessionContext
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)`

### Environment Information

- **Operating System**: Ubuntu under WSL
- **Micronaut Version:** 3.7.1
- **JDK Version:** 11

### Example Application

https://micronaut.io/launch?type=DEFAULT&name=demo-jdbc&package=com.example&javaVersion=JDK_11&lang=JAVA&build=GRADLE&test=JUNIT&features=spring-data-jdbc&version=3.7.1

**Comment by wetted** [comment]
I have a version that works, but it doesn't feel entirely the right way to fix this. I had to do two things:
1) add two dependencies: 
```
implementation("io.micronaut.data:micronaut-data-tx-hibernate")
implementation("javax.persistence:javax.persistence-api:2.2")
``` 
2) add a JPA entity, for example.
```
package com.example;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Entity
public class Person {
    @Id
    @GeneratedValue
    private Long id;
    ...
}
```

ping @dstepanov Any thoughts. This is for Micronaut 3.x. It's not an issue with Micronaut 4.

**Comment by wetted** [comment]
via Discord, @radovanradic wrote
> this is fixed in Micronaut 4.x and for 3.10.x maybe as a workaround this can be used 
> ```
> implementation("io.micronaut.data:micronaut-data-spring") {
>         exclude group: "io.micronaut.sql", module: "micronaut-hibernate-jpa"
>         exclude group: "io.micronaut.sql", module: "micronaut-hibernate-jpa-spring"
>     }
> ```
> the issue is that spring-data-jdbc feature included micronaut-data-spring and it included in 3.10.x micronaut-sql-hibernate-jpa which needed MicronautSessionContext which is in micronaut-data-tx-hibernate and that's not included
> Basically, jpa and hibernate should not be included for spring-jdbc feature but the issue was with micronaut-data-spring which is now fixed in 4.x
